backend/analytics_server/mhq/service/external_integrations_service.py

```diff
@@ -57,10 +63,66 @@ def get_repo_workflows(self, gh_org_name: str, gh_org_repo_name: str):
         except GithubException as e:
             raise e
 
+    def get_gitlab_groups(self) -> List[Dict]:
+        gitlab_api_service = GitlabApiService(self.access_token, self.custom_domain)
+        try:
+            groups: List[Dict] = gitlab_api_service.get_groups()
+        except Exception as e:
+            raise e
+
+        return groups
+
+    def get_gitlab_group_projects(
+        self, group_id: str, page_size: int, page: int
+    ) -> List[GitlabRepo]:
+        gitlab_api_service = GitlabApiService(self.access_token, self.custom_domain)
+        try:
+            projects: List[Dict] = gitlab_api_service.get_group_projects(
+                group_id, page_size, page
+            )
+        except Exception as e:
+            raise e
+
+        return projects
+
+    def get_gitlab_user_projects(self, page_size: int, page: int) -> List[GitlabRepo]:
+        gitlab_api_service = GitlabApiService(self.access_token, self.custom_domain)
+        try:
+            user: GitlabUser = gitlab_api_service.get_authenticated_user()
+            user_id: str = user.meta.get("id")
+            projects: List[GitlabRepo] = gitlab_api_service.get_user_projects(
+                user_id, page, page_size
+            )
+        except Exception as e:
+            raise e
+
+        return projects
+
 
 def get_external_integrations_service(
     org_id: str, user_identity_provider: UserIdentityProvider
 ):
+    def _get_custom_gitlab_domain() -> Optional[str]:
+        DEFAULT_DOMAIN = "gitlab.com"
+        core_repo_service = CoreRepoService()
+        integrations = core_repo_service.get_org_integrations_for_names(
+            org_id, [UserIdentityProvider.GITLAB.value]
+        )
+
+        gitlab_domain = (
+            integrations[0].provider_meta.get("custom_domain")
+            if integrations[0].provider_meta
+            else None
+        )
+
+        if not gitlab_domain:
+            LOG.warn(
+                f"Custom domain not found for intergration for org {org_id} and provider {UserIdentityProvider.GITLAB.value}"
+            )
+            return DEFAULT_DOMAIN
```

Comment:
Would break for someone with github integration, and you are calling it irrespective of the provider